### PR TITLE
feat: update to solidity 0.8, OZ 4.1 and ytoken fixs

### PR DESCRIPTION
**Pull request by steffenix**

### CODE REVIEW

File: contracts/yToken.sol [hunk 1]
```diff
@@ -1,34 +1,27 @@
...
-        VaultAPI _bestVault = bestVault();
-        return _bestVault.symbol();
+        return string(abi.encodePacked("y", IERC20Metadata(address(token)).symbol()));
     }
 
     function decimals() external view returns (uint256) {
-        VaultAPI _bestVault = bestVault();
-        return _bestVault.decimals();
+        return IERC20Metadata(address(token)).decimals();
     }
```
Comment (fubuloubu) [inline contracts/yToken.sol:25]: Can these changes be brought out into another PR? Basically, define `IERC20Metadata` interface above and use that until "upgrading" to OZ version

**Review by fubuloubu — APPROVED** [review]
Don't merge until v1.0.0 is ready to release

File: contracts/yToken.sol (hunk 1 above)
Comment (steffenix) [reply contracts/yToken.sol:25]: ok will do

**Comment by steffenix** [comment]
Closing this for now
